### ISSUE TITLE
Custom property declaration - double colon allowed

### ISSUE THREAD

**Issue by mcoker** [stylelint/stylelint#7360]
### What minimal example or steps are needed to reproduce the bug?

A codesandbox demonstrating the issue here - https://codesandbox.io/p/devbox/eager-snowflake-c5393y?file=%2Fsrc%2Findex.css%3A6%2C1

A custom property declaration with multiple colons between the property and value does not report as an error. Eg, `--color:: white;`. However a regular property declaration with multiple colons does report as an error. Eg, `color:: var(--color);` 

### What minimal configuration is needed to reproduce the bug?

I don't know that it's necessary, but it can be reproduced with `"extends": ["stylelint-config-recommended"],`

### How did you run Stylelint?

https://codesandbox.io/p/devbox/eager-snowflake-c5393y?file=%2F.stylelintrc.json%3A2%2C3-2%2C47

### Which Stylelint-related dependencies are you using?

```
    "stylelint": "15.11.0",
    "stylelint-config-recommended": "13.0.0",
```

### What did you expect to happen?

Multiple colons in a custom property declaration would report as an error just like a regular property declaration.

### What actually happened?

Custom property declaration doesn't report as an error, only regular property declarations do.

**Comment by jeddy3** [comment]
@mcoker Thanks for the report and for using the template.

I believe `--color:: white;` is valid CSS as you can do all sorts of stuff with custom properties. From [the spec](https://www.w3.org/TR/css-variables-1/#syntax):

> The allowed syntax for [custom properties](https://www.w3.org/TR/css-variables-1/#custom-property) is extremely permissive. The [<declaration-value>](https://www.w3.org/TR/css-syntax-3/#typedef-declaration-value) production matches any sequence of one or more tokens, so long as the sequence does not contain [<bad-string-token>](https://www.w3.org/TR/css-syntax-3/#typedef-bad-string-token), [<bad-url-token>](https://www.w3.org/TR/css-syntax-3/#typedef-bad-url-token), unmatched [<)-token>](https://www.w3.org/TR/css-syntax-3/#tokendef-close-paren), [<]-token>](https://www.w3.org/TR/css-syntax-3/#tokendef-close-square), or [<}-token>](https://www.w3.org/TR/css-syntax-3/#tokendef-close-curly), or top-level [<semicolon-token>](https://www.w3.org/TR/css-syntax-3/#typedef-semicolon-token) tokens or [<delim-token>](https://www.w3.org/TR/css-syntax-3/#typedef-delim-token) tokens with a value of "!".

We'll unlikely want to write a built-in rule to disallow this pattern. However, you can use the [declaration-property-value-disallowed-list](https://stylelint.io/user-guide/rules/declaration-property-value-disallowed-list/) rule to disallow this pattern in your codebase:

```json
{
  "extends": [
    "stylelint-config-standard"
  ],
  "rules": {
    "declaration-property-value-disallowed-list": {
      "/^--/": ["/:/"]
    }
  }
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcCBOEIBcAEwMFoeQBsIl54MB3ACwEs0YBuDAXxABoQAzagmAOQEMAtnEQwAHkIAOPAHRgAzvLbgIAOy4BzBCGAAdVRgy6Q4uqqjzjZANr7Dh4-LQBPHgWqq0+NZpxP+5vxIUMZ2GAC6rGHGSACuPJYIWGEOILBgBEH8aNRqOJIokjBILjgAbvwEsTA4UNTylUTkMFA47k5WyQb2qQD0AHp4vZ22IL3wwyDhKcxhTPos7JDq1BoAYsSC2doAVvJqyrCSSoh63Y4ubh5oncaZdB1s0SBOrjDunt4rGn5oAVBBEJJO7ZGCPBYgJhAA).

I'll label the issue as discussion, in case anyone else has anything to add.

**Comment by ybiquitous** [comment]
This is an interesting case. I didn't know double colon interpretation differs between a typical property and a custom one.

I agree with configuring `declaration-property-value-disallowed-list` (as @jeddy3 suggested) instead of adding a new rule because this case seems uncommon to me.

**Comment by jeddy3** [comment]
> I didn't know double colon interpretation differs between a typical property and a custom one.

Yep, custom property values are very permissive. I believe all of these are valid:

```css
:root {
  --custom-prop: ;
  --custom-prop::::;
  --custom-prop: if(x > 5) this.width = 10;
  --custom-prop: ¯_(ツ)_/¯;
}
```
